Build error while compiling with LWIP2 with IPv6

### Basic Infos

- [X] This issue complies with the [issue POLICY doc](https://github.com/esp8266/Arduino/blob/master/POLICY.md).
- [X] I have read the documentation at [readthedocs](https://arduino-esp8266.readthedocs.io/en/latest) and the issue is not addressed there.
- [X] I have tested that the issue is present in current master branch (aka latest git).
- [X] I have searched the issue tracker for a similar issue.
- [X] [N/A] If there is a stack dump, I have decoded it.
- [X] I have filled out all fields below.

#### Platform

- Hardware:				ESP-12
- Core Version:			Master (8049543e989c5b23d0d0d4f3829e385f08307d36) and 2.5.0-beta2
- Development Env:		Arduino IDE
- Operating System:		Ubuntu

### Settings in IDE

- Module:  			Adafruit Feather HUZZAH ESP8266. Same with Generic ESP8266 module
- Flash Mode:			?? No idea where to find this. If needed, please let me know
- Flash Size: 			4MB
- lwip Variant:  		v2 IPv6 Lower Memory && v2 IPv6 Higher Bandwidth
- Reset Method:		?? No idea where to find this. If needed, please let me know
- Flash Frequency:		40MHz
- CPU Frequency:		80MHz
- Upload Using:		SERIAL
- Upload Speed:   		115200

### Problem Description

When building the provided IPv6 example in this repo for LWip 2.0 with IPv6 support, some build errors arrise. From the source code i could not find a direct cause, so i opened it here for someone to look at. Should be reproducable with newest arduino IDE (1.8.8) and master or 2.5.0-beta2 from this repo.

### [MCVE](https://stackoverflow.com/help/mcve) Sketch
Sketch: Provided ESP6266WiFi/IPv6/IPv6.ino example in this repo.

### Debug Messages

```
In file included from /home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/libraries/ESP8266WiFi/examples/IPv6/IPv6.ino:22:0:
/home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/cores/esp8266/AddrList.h: In member function 'IPAddress esp8266::AddressListImplementation::netifWrapper::addr() const':
/home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/cores/esp8266/AddrList.h:113:61: error: conversion from 'const ip_addr_t* {aka const ip_addr*}' to 'IPAddress' is ambiguous
     IPAddress addr () const         { return ipFromNetifNum(); }
                                                             ^
/home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/cores/esp8266/AddrList.h:113:61: note: candidates are:
In file included from /home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/libraries/ESP8266WiFi/src/ESP8266WiFi.h:31:0,
                 from /home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/libraries/ESP8266WiFi/examples/IPv6/IPv6.ino:19:
/home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/cores/esp8266/IPAddress.h:79:9: note: IPAddress::IPAddress(int) <near match>
         IPAddress(int address) { ctor32(address); }
         ^
/home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/cores/esp8266/IPAddress.h:79:9: note:   no known conversion for argument 1 from 'const ip_addr_t* {aka const ip_addr*}' to 'int'
/home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/cores/esp8266/IPAddress.h:78:9: note: IPAddress::IPAddress(u32_t) <near match>
         IPAddress(u32_t address) { ctor32(address); }
         ^
/home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/cores/esp8266/IPAddress.h:78:9: note:   no known conversion for argument 1 from 'const ip_addr_t* {aka const ip_addr*}' to 'u32_t {aka long unsigned int}'
/home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/cores/esp8266/IPAddress.h:77:9: note: IPAddress::IPAddress(uint32_t) <near match>
         IPAddress(uint32_t address) { ctor32(address); }
         ^
/home/jk-5/.arduino15/packages/esp8266/hardware/esp8266/2.5.0-beta2/cores/esp8266/IPAddress.h:77:9: note:   no known conversion for argument 1 from 'const ip_addr_t* {aka const ip_addr*}' to 'uint32_t {aka unsigned int}'
exit status 1
Error compiling for board Adafruit Feather HUZZAH ESP8266.
```

--- a/cores/esp8266/IPAddress.cpp
+++ b/cores/esp8266/IPAddress.cpp
@@ -27,11 +27,6 @@ IPAddress::IPAddress(const IPAddress& from)
     ip_addr_copy(_ip, from._ip);
 }
 
-IPAddress::IPAddress(const ip_addr_t* from)
-{
-    ip_addr_copy(_ip, *from);
-}
-
 IPAddress::IPAddress() {
     _ip = *IP_ANY_TYPE; // lwIP's v4-or-v6 generic address
 }
@@ -191,6 +186,11 @@ const IPAddress INADDR_NONE(255,255,255,255);
 /**************************************/
 
 #if LWIP_IPV6
+
+IPAddress::IPAddress(const ip_addr_t* from)
+{
+    ip_addr_copy(_ip, *from);
+}
 
 bool IPAddress::fromString6(const char *address) {
     // TODO: test test test

--- a/cores/esp8266/IPAddress.cpp
+++ b/cores/esp8266/IPAddress.cpp
@@ -27,6 +27,11 @@ IPAddress::IPAddress(const IPAddress& from)
     ip_addr_copy(_ip, from._ip);
 }
 
+IPAddress::IPAddress(const ip_addr_t* from)
+{
+    ip_addr_copy(_ip, *from);
+}
+
 IPAddress::IPAddress() {
     _ip = *IP_ANY_TYPE; // lwIP's v4-or-v6 generic address
 }

--- a/cores/esp8266/IPAddress.h
+++ b/cores/esp8266/IPAddress.h
@@ -73,7 +73,6 @@ class IPAddress: public Printable {
         // Constructors
         IPAddress();
         IPAddress(const IPAddress& from);
-        IPAddress(const ip_addr_t* from);
         IPAddress(uint8_t first_octet, uint8_t second_octet, uint8_t third_octet, uint8_t fourth_octet);
         IPAddress(uint32_t address) { ctor32(address); }
         IPAddress(u32_t address) { ctor32(address); }
@@ -163,6 +162,8 @@ class IPAddress: public Printable {
         bool isLocal () const { return ip_addr_islinklocal(&_ip); }
 
 #if LWIP_IPV6
+
+        IPAddress(const ip_addr_t* from);
 
         uint16_t* raw6()
         {

--- a/cores/esp8266/IPAddress.h
+++ b/cores/esp8266/IPAddress.h
@@ -73,6 +73,7 @@ class IPAddress: public Printable {
         // Constructors
         IPAddress();
         IPAddress(const IPAddress& from);
+        IPAddress(const ip_addr_t* from);
         IPAddress(uint8_t first_octet, uint8_t second_octet, uint8_t third_octet, uint8_t fourth_octet);
         IPAddress(uint32_t address) { ctor32(address); }
         IPAddress(u32_t address) { ctor32(address); }

--- a/libraries/ESP8266WiFi/examples/IPv6/IPv6.ino
+++ b/libraries/ESP8266WiFi/examples/IPv6/IPv6.ino
@@ -98,7 +98,11 @@ void setup() {
   Serial.println();
   Serial.println(ESP.getFullVersion());
 
-  Serial.printf("IPV6 is%s enabled\n", LWIP_IPV6 ? emptyString.c_str() : " NOT");
+#if LWIP_IPV6
+  Serial.printf("IPV6 is enabled\n");
+#else
+  Serial.printf("IPV6 is not enabled\n");
+#endif
 
   WiFi.mode(WIFI_STA);
   WiFi.begin(STASSID, STAPSK);

